src/sql/workbench/parts/query/browser/gridPanel.ts

```diff
@@ -419,7 +419,23 @@ export abstract class GridTableBase<T> extends Disposable implements IView {
 		this.scrolled = false;
 	}
 
-	private build(): void {
+	// actionsOrientation controls the orientation (horizontal or vertical) of the actionBar
+	private build(actionsOrientation?: ActionsOrientation): void {
+
+		// Default is VERTICAL
+		if (isUndefinedOrNull(actionsOrientation)) {
+			actionsOrientation = ActionsOrientation.VERTICAL;
+		}
+
+		let actionBarContainer = document.createElement('div');
+
+		// Create a horizontal actionbar if orientation passed in is HORIZONTAL
+		if (actionsOrientation === ActionsOrientation.HORIZONTAL) {
+			actionBarContainer.style.width = '100%';
+			actionBarContainer.style.display = 'flex';
+			this.container.appendChild(actionBarContainer);
+		}
```

Comment:
It'd be nice to put this down with the other block of code for the vertical alignment

Reply:
Since the ordering already matters when the container is added, I think I like this a bit better. Maybe it's a bit more readable with the styling living in CSS now?